### DISCUSSION TITLE
HTTP serve a directory with dynamic runtime path

### DISCUSSION THREAD

**Discussion by PaulHax** [Kitware/trame#536] [Q&A] [answered]
For the NRTK-Explorer trame app, I'd like to HTTP serve a directory of images that is changed at runtime.  The `serve` module is enabled at startup to dish out the apps custom Vue component bundle, so later calls to `server.enable_module({"serve": {"dataset": str(dataset_dir.absolute())}})` seem to be ignored.
https://github.com/Kitware/nrtk-explorer/pull/72/files#diff-7689df3670f1c5e555401b178b486a23f4ed6f9d418aaaeaf810204bdf30e9bcR29

Wondering how to HTTP serve another directory behond the initial?

**Comment by jourdain** [comment, marked as answer]
enable modules needs to happen "before the server start". 

Another option is to create a dynamic endpoint that [can be registered at the beginning](https://github.com/banesullivan/trame-tileserver/blob/main/app.py#L80-L83) but where its behavior can be updated at runtime. 

But if we go that route, we will run into an issue if we deploy such app in a docker context as at that point each trame server are just handling the WebSocket and none of the HTTP static content delivery.
